pkgs/applications/networking/instant-messengers/signal-desktop/generic.nix

```diff
@@ -236,7 +236,7 @@ stdenv.mkDerivation rec {
 
     # Fix the desktop link
     substituteInPlace $out/share/applications/${pname}.desktop \
-      --replace "/opt/${dir}/${pname}" $out/bin/${pname} \
+      --replace-fail "/opt/${dir}/${pname}" ${meta.mainProgram} \
```

Comment:
Why is mainProgram defaulting to pname here? The netire point of adding mainProgram was to no longer default to pname....

Reply:
It's because the different variants of this package have different binary names, but I do agree that this isn't good. I can make a separate PR to fix this